Sources/Amplitude/Amplitude.m

```diff
@@ -1074,19 +1073,40 @@ - (void)enterForeground {
 #endif
     [self runOnBackgroundQueue:^{
         // Fetch the data ingestion endpoint based on current device's geo location.
-        
         [self refreshDynamicConfig];
         [self startOrContinueSessionNSNumber:now];
+        self->_eventsBuffer =  [self mergeBufferWithFileStorage:EVENTS];
+        self->_identifyBuffer = [self mergeBufferWithFileStorage:IDENTIFY_EVENT];
         self->_inForeground = YES;
-        self->_eventsBuffer = [AMPStorage getEventsFromDisk:[AMPStorage getDefaultEventsFile:self.instanceName]];
-        self->_identifyBuffer = [AMPStorage getEventsFromDisk:[AMPStorage getDefaultIdentifyFile:self.instanceName]];
         if ([self->_eventsBuffer count] > 0 || [self->_identifyBuffer count] > 0) {
             [self uploadEvents];
-            
         }
     }];
 }
 
+- (NSMutableArray *)mergeBufferWithFileStorage:(NSString *)eventType {
+    NSMutableArray *mergedEvent =  [[NSMutableArray alloc] init];
+    NSMutableArray *buffer = [[NSMutableArray alloc] init];
+    long long maxSequenceNumber = 0;
+    if ([eventType isEqualToString:IDENTIFY_EVENT]) {
+        mergedEvent = [AMPStorage getEventsFromDisk:[AMPStorage getDefaultIdentifyFile:self.instanceName]];
+        maxSequenceNumber = self->_maxIdentifySequenceNumber;
+        buffer = self->_identifyBuffer;
+    } else {
+        mergedEvent = [AMPStorage getEventsFromDisk:[AMPStorage getDefaultEventsFile:self.instanceName]];
+        maxSequenceNumber = self->_maxEventSequenceNumber;
+        buffer = self->_eventsBuffer;
+    }
+    for (NSDictionary *event in buffer) {
```

Comment:
Actually I guess from this line can be a separate function, and we don't have to use the event type to tell the difference,
could just be something like
```
events = [AMPStorage getEventsFromDisk:[AMPStorage getDefaultEventsFile:self.instanceName]];
self->_eventsBuffer = [mergeBufferWithEvents events buffer:self->eventsBuffer];
identifyEvents = ...
self->_idenfityBuffer = ...

(NSMutableArray *)mergeBufferWithEvents: events, buffer
```